providers/dns/dynu/internal/client_test.go

```diff
@@ -0,0 +1,317 @@
+			pattern: "/dns/record/_acme-challenge.ldez.freeddns.org",
+			status:  http.StatusOK,
+			file:    "./fixtures/get_records.json",
+			expected: expected{
+				records: []DNSRecord{{
+					ID:         6041417,
+					Type:       "TXT",
+					DomainID:   9007481,
+					DomainName: "ldez.freeddns.org",
+					NodeName:   "_acme-challenge",
+					Hostname:   "_acme-challenge.ldez.freeddns.org",
+					State:      true,
+					Content:    `_acme-challenge.ldez.freeddns.org. 300 IN TXT "txt_txt_txt_txt_txt_txt_txt"`,
+					TextData:   "txt_txt_txt_txt_txt_txt_txt",
+					TTL:        300,
+				},
+					{
+						ID:         6041422,
+						Type:       "TXT",
+						DomainID:   9007481,
+						DomainName: "ldez.freeddns.org",
+						NodeName:   "_acme-challenge",
+						Hostname:   "_acme-challenge.ldez.freeddns.org",
+						State:      true,
+						Content:    `_acme-challenge.ldez.freeddns.org. 300 IN TXT "txt_txt_txt_txt_txt_txt_txt_2"`,
+						TextData:   "txt_txt_txt_txt_txt_txt_txt_2",
+						TTL:        300,
+					},
+				},
+			},
+		},
+		{
+			desc:    "empty",
+			pattern: "/dns/record/_acme-challenge.ldez.freeddns.org",
+			status:  http.StatusOK,
+			file:    "./fixtures/get_records_empty.json",
+			expected: expected{
+				records: []DNSRecord{},
+			},
+		},
+		{
+			desc:    "invalid",
+			pattern: "/dns/record/_acme-challenge.ldez.freeddns.org",
+			status:  http.StatusNotImplemented,
+			file:    "./fixtures/get_records_invalid.json",
+			expected: expected{
+				error: "API error: 501: Argument Exception: Invalid.",
+			},
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+
+			client := setupTest(http.MethodGet, test.pattern, test.status, test.file)
+
+			records, err := client.GetRecords("_acme-challenge.ldez.freeddns.org", "TXT")
+
+			if test.expected.error != "" {
+				assert.EqualError(t, err, test.expected.error)
+				return
+			}
+
+			require.NoError(t, err)
+
+			assert.NotNil(t, records)
+			assert.Equal(t, test.expected.records, records)
+		})
+	}
+
+	t.Skip("now")
+	client := NewClient()
+
+	// {"statusCode":200,"dnsRecords":[{"id":6041417,"domainId":9007481,"domainName":"ldez.freeddns.org","nodeName":"_acme-challenge","hostname":"_acme-challenge.ldez.freeddns.org","recordType":"TXT","ttl":300,"state":true,"content":"_acme-challenge.ldez.freeddns.org. 300 IN TXT \"txt_txt_txt_txt_txt_txt_txt\"","updatedOn":"2020-03-10T04:00:36.923","textData":"txt_txt_txt_txt_txt_txt_txt"},{"id":6041422,"domainId":9007481,"domainName":"ldez.freeddns.org","nodeName":"_acme-challenge","hostname":"_acme-challenge.ldez.freeddns.org","recordType":"TXT","ttl":300,"state":true,"content":"_acme-challenge.ldez.freeddns.org. 300 IN TXT \"txt_txt_txt_txt_txt_txt_txt_2\"","updatedOn":"2020-03-10T04:03:17.563","textData":"txt_txt_txt_txt_txt_txt_txt_2"}]}
+
+	// {"statusCode":200,"dnsRecords":[]}
+
+	// {"statusCode":501,"type":"Argument Exception","message":"Invalid."}
```

Comment:
These comments seem already live as test fixture, so they can be removed as well.